libs/server/Resp/Vector/RespServerSessionVectors.cs

```diff
@@ -1004,11 +1004,47 @@ private bool NetworkVINFO<TGarnetApi>(ref TGarnetApi storageApi)
                 return AbortWithErrorMessage("ERR Vector Set (preview) commands are not enabled");
             }
 
-            // TODO: implement!
+            if (parseState.Count != 1)
+            {
+                return AbortWithWrongNumberOfArguments("VINFO");
+            }
 
-            while (!RespWriteUtils.TryWriteDirect(CmdStrings.RESP_OK, ref dcurr, dend))
-                SendAndReset();
+            var key = parseState.GetArgSliceByRef(0);
+            var res = storageApi.VectorSetInfo(key, out VectorQuantType quantType, out uint vectorDimensions, out uint reducedDimensions, out uint buildExplorationFactor, out uint numLinks, out long size);
+            if (res == GarnetStatus.NOTFOUND)
+            {
+                WriteNull();
```

Comment:
This is subtly wrong - see [`VINFO` Returns](https://redis.io/docs/latest/commands/vinfo/#return-information).

If `RespProtocolVersion == 3` this should return the null (`_\r\n`), but in RESP 2 it writes the "null array" (`*-1\r\n`).  This is wrapped up in `WriteNullArray()`, so just call that instead.

 Many (but not all) libraries will paper over that mismatch, so understandable it got missed.